Bottom: d76efaf70a63c95b36e280145249e139f1702731
Top:    16ce99869f4fdc5a471cfe4f5c61f5c63b26c71c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 03:14:57 +0000

Reduce the redundant work done in node_label


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 35400a8..043db86 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -424,13 +424,16 @@ node_label (VNODE * pn)
 	  if (l->side == 'P')
 	    {
 	      region = INSIDE;
+#if 0
 	      if (l->parent->prev->point[0] == pn->prev->point[0] &&
 		  l->parent->prev->point[1] == pn->prev->point[1])
 		{
 		  LABEL_NODE (pn->prev, SHARED);	/* incoming is shared */
 		  pn->prev->shared = l->parent->prev;
 		}
-	      else if (l->parent->prev->point[0] == pn->next->point[0] &&
+	      else
+#endif
+                   if (l->parent->prev->point[0] == pn->next->point[0] &&
 		       l->parent->prev->point[1] == pn->next->point[1])
 		{
 		  LABEL_NODE (pn, SHARED2);	/* outgoing is shared2 */
@@ -446,12 +449,14 @@ node_label (VNODE * pn)
 		  LABEL_NODE (pn, SHARED);
 		  pn->shared = l->parent;
 		}
+#if 0
 	      else if (l->parent->next->point[0] == pn->prev->point[0] &&
 		       l->parent->next->point[1] == pn->prev->point[1])
 		{
 		  LABEL_NODE (pn->prev, SHARED2);	/* outgoing is shared2 */
 		  pn->prev->shared = l->parent;
 		}
+#endif
 	    }
 	}
       else
@@ -461,6 +466,8 @@ node_label (VNODE * pn)
 	    v = l->parent->prev;
 	  else
 	    v = l->parent;
+          if (v != pn)
+            continue;
 	  if (NODE_LABEL (v) != SHARED && NODE_LABEL (v) != SHARED2)
 	    {
 #ifdef DEBUG_LABEL
@@ -487,8 +494,8 @@ node_label (VNODE * pn)
 #ifdef DEBUG_LABEL
   DEBUGP ("\n");
 #endif
-  assert (NODE_LABEL (pn) != UNKNWN && NODE_LABEL (pn->prev) != UNKNWN);
-  if (NODE_LABEL (pn) == UNKNWN || NODE_LABEL (pn->prev) == UNKNWN)
+  assert (NODE_LABEL (pn) != UNKNWN /* && NODE_LABEL (pn->prev) != UNKNWN */);
+  if (NODE_LABEL (pn) == UNKNWN /* || NODE_LABEL (pn->prev) == UNKNWN */)
     return UNKNWN;
   if (NODE_LABEL (pn) == INSIDE || NODE_LABEL (pn) == OUTSIDE)
     return NODE_LABEL (pn);
@@ -1142,14 +1149,17 @@ label_contour (PLINE * a)
           continue;
         }
 
+#if 0
       if (NODE_LABEL (cur) != UNKNWN)
         {
+          fprintf (stderr, "Not expected...\n");
           /* This might get hit if the last cross vertex labelling
            * pass was a bit keen, and labeled past its own edges?
            */
           label = NODE_LABEL (cur);
           continue;
         }
+#endif
 
       if (label == INSIDE || label == OUTSIDE)
         {
